src/lib.rs

```diff
@@ -0,0 +1,292 @@
+use std::collections::HashSet;
+use std::fmt;
+
+use guest::prelude::*;
+use kubewarden_policy_sdk::wapc_guest as guest;
+
+use k8s_openapi::api::core::v1::{EnvVar, PodSpec};
+extern crate kubewarden_policy_sdk as kubewarden;
+use encoding::all::ASCII;
+use encoding::{DecoderTrap, Encoding};
+use kubewarden::{protocol_version_guest, request::ValidationRequest, validate_settings};
+use rusty_hog_scanner::{SecretScanner, SecretScannerBuilder};
+use std::string::String;
+
+mod settings;
+use settings::Settings;
+
+#[no_mangle]
+pub extern "C" fn wapc_init() {
+    register_function("validate", validate);
+    register_function("validate_settings", validate_settings::<Settings>);
+    register_function("protocol_version", protocol_version_guest);
+}
+
+/// Represents a secret that has been found in an env var
+#[derive(Eq, Hash, PartialEq, Debug)]
+struct SecretRejected {
+    /// nome of the container where the secret was found
+    container: String,
+    /// reason of rejection. It describes the secret that it was found
+    reason: String,
+    /// key of the env var
+    key: String,
+}
```

Comment:
Thinking more about the code change I proposed, the one that keep track of how many times the same rejection reason was found, I think it would be better to extend this struct to include also the occurrence counter

Reply:
@flavio I removed the `ASCII decode` code since we don't need it. I also added `base64` here https://github.com/raulcabello/env-variable-secrets-scanner-policy/commit/54d11aafca309493ae2022a1c1e11afad8dc9340